BaseClasses.py

```diff
@@ -1087,14 +1087,16 @@ class Spoiler:
     playthrough: Dict[str, Union[List[str], Dict[str, str]]]  # sphere "0" is list, others are dict
     unreachables: Set[Location]
     paths: Dict[str, List[Union[Tuple[str, str], Tuple[str, None]]]]  # last step takes no further exits
+    level: int
 
-    def __init__(self, multiworld: MultiWorld) -> None:
+    def __init__(self, multiworld: MultiWorld, level: int = 0) -> None:
```

Comment:
Wondering if the default of `0` is actually a good idea - i would expect the default to be full spoiler.

In theory we could also reference `settings.Spoiler` for the typing, but integrating proper types into Generate is probably a bigger task in itself, because you'd want to upcast the command line args to meet settings' types.

Reply:
Maybe have an explicit `IntEnum` and use it in both so these levels can have a meaningful repr?